abe/fame.go

```diff
@@ -120,6 +121,39 @@ func (a *FAME) Encrypt(msg string, msp *MSP, pk *FAMEPubKey) (*FAMECipher, error
 
 	}
 
+	// msg is encrypted using CBC, with a random key that is encapsulated
+	// with GPSW
+	_, keyGt, err := bn256.RandomGT(rand.Reader)
+	if err != nil {
+		return nil, err
+	}
+	keyCBC := sha256.Sum256([]byte(keyGt.String()))
+
+	c, err := aes.NewCipher(keyCBC[:])
+	if err != nil {
+		return nil, err
+	}
+
+	iv := make([]byte, c.BlockSize())
+	encrypterCBC := cbc.NewCBCEncrypter(c, iv)
+
+	msgByte := []byte(msg)
+
+	// message is padded according to pkcs7 standard
+	padLen := c.BlockSize() - (len(msgByte) % c.BlockSize())
+	msgPad := make([]byte, len(msgByte)+padLen)
+	for i := 0; i < len(msgPad); i++ {
+		if i < len(msgByte) {
+			msgPad[i] = msgByte[i]
+		} else {
+			msgPad[i] = byte(padLen)
+		}
+	}
```

Comment:
I think you can copy the `msgByte` contents to `msgPad` with [copy builtin](https://golang.org/pkg/builtin/#copy).
Additionally, why is `msgPad[i]` (for `i >= len(msgByte)`) set to `byte(padLen)`? That just performs a %256 operation, doesn't it?

Reply:
I changed the copy part. CBC encryption demands that the encrypted message is of length multiple of 16 (in bytes). Hence message must be padded. PKCS7 standard prescribes that we add (16 - (len(msg) mod 16)) bytes, each containing this value this value (as a byte).

Reply:
Good to know. I wasn't sure where the padding value was coming from.